Cython/Compiler/Scanning.py

```diff
@@ -538,3 +552,29 @@ def exit_async(self):
             self.keywords.discard('async')
             if self.sy in ('async', 'await'):
                 self.sy, self.systring = IDENT, self.context.intern_ustring(self.sy)
+
+@contextmanager
+def tentatively_scan(scanner):
```

Comment:
Since we're doing a lot of attribute accesses here, typing the `scanner` argument seems worth it.
```suggestion
@contextmanager
@cython.locals(scanner=Scanner)
def tentatively_scan(scanner):
```